src/plugins/signoff/index.js

```diff
@@ -52,20 +84,59 @@ class SignoffButton extends React.Component {
       return null;
     }
 
+    const {status="work-in-progress"} = collectionState.data;
+    let action;
+    let label;
+    if (status === "to-review") {
+      return (
+        <span>
+          <a className="btn btn-success"
+             href="#"
+             onClick={(event) => {
+               event.preventDefault();
+               dispatch(approveChanges());
+             }}><i className="glyphicon glyphicon-ok"></i> Approve changes</a>
+          <a className="btn btn-danger"
+             href="#"
+             onClick={(event) => {
+               event.preventDefault();
+               dispatch(declineChanges());
+             }}><i className="glyphicon glyphicon-remove"></i> Decline changes</a>
+
+        </span>
+      );
+    } else if (status === "signed") {
+      action = approveChanges();
```

Comment:
Could we find an alternative way to define and handle these? The code path is slightly confusing to follow